app_test.go

```diff
@@ -420,9 +427,15 @@ func TestOptions(t *testing.T) {
 		spy := new(fxlog.Spy)
 		New(
 			Provide(&bytes.Buffer{}), // error, not a constructor
-			Logger(spy),
+			WithLogger(spy),
 		)
-		assert.Contains(t, spy.String(), "must provide constructor function")
+		assert.Contains(t, spy.String(), "error encountered while applying options")
+		fields := spy.Fields()
+		assert.Len(t, fields, 1)
+		s := fields[0]
+		assert.Equal(t, "error", s.Key)
+		assert.Equal(t, zapcore.ErrorType, s.Type)
+		assert.Contains(t, fmt.Sprintf("%v", s.Interface), "must provide constructor function")
```

Comment:
can do fmt.Sprint instead of `fmt.Sprintf` with a `%v`

```suggestion
		assert.Contains(t, fmt.Sprint(s.Interface), "must provide constructor function")
```